### PR TITLE
Fix Editor Play-Mode RequestNetSpawnable Error

### PR DESCRIPTION
**Pull request by AMZN-Gene**
- Fixes an error seen when using editor play-mode with multiplayer by making sure net spawnables aren't spawned until net agent type is set; once the editor becomes a client, then it's fine to create network spawnables.
- Misc code improvement found while debugging
- AutomatedTest added which will run the editor and make sure the error doesn't occur. The test also does a simple check to ensure net-level-entities can properly run scripts (right now it's only checking server side since there's issues with client, but will be expanded to handle client once the issue is resolved)
- Also tested by running Multiplayer Sample

### CODE REVIEW

File: AutomatedTesting/Gem/PythonTests/Multiplayer/tests/Multiplayer_SimpleNetworkLevelEntity.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    with Tracer() as section_tracer:
+        # 2) Enter game mode
+        helper.multiplayer_enter_game_mode(TestSuccessFailTuples.enter_game_mode, player_prefab_path.lower())
+
+        # 3) Make sure the network player was spawned
+        player_id = general.find_game_entity(player_prefab_name)
+        Report.critical_result(TestSuccessFailTuples.find_network_player, player_id.IsValid())
+
+        # 4) Check the editor logs for network spawnable errors
+        ATTEMPTING_INVALID_NETSPAWN_WAIT_TIME_SECONDS = 1.0  # The editor will try to net-spawn its networked level entity before it's even a client. Make sure this doesn't happen.
```
Comment (lmbr-pip) [inline AutomatedTesting/Gem/PythonTests/Multiplayer/tests/Multiplayer_SimpleNetworkLevelEntity.py:61]: nit: Is 1.0s long enough on slow machines?

File: AutomatedTesting/Gem/PythonTests/Multiplayer/tests/Multiplayer_SimpleNetworkLevelEntity.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        helper.fail_if_log_line_found('NetworkEntityManager', "RequestNetSpawnableInstantiation: Requested spawnable Root.network.spawnable doesn't exist in the NetworkSpawnableLibrary. Please make sure it is a network spawnable", section_tracer.errors, ATTEMPTING_INVALID_NETSPAWN_WAIT_TIME_SECONDS)
+
+        # 5) Ensure the script graph attached to the level entity is running on both client and server
+        SCRIPTGRAPH_ENABLED_WAIT_TIME_SECONDS = 0.25
+        # Check Server
+        helper.succeed_if_log_line_found('EditorServer', "Script: SimpleNetworkLevelEntity: On Graph Start", section_tracer.prints, SCRIPTGRAPH_ENABLED_WAIT_TIME_SECONDS)
+        # Check Editor/Client (Uncomment once script asset preload is working properly LYN-9136)
```
Comment (lmbr-pip) [inline AutomatedTesting/Gem/PythonTests/Multiplayer/tests/Multiplayer_SimpleNetworkLevelEntity.py:68]: Lets not check in commented out code, prefer to create issue for next steps.

**Review by lmbr-pip — COMMENTED** [review]
LG overall

File: AutomatedTesting/Gem/PythonTests/Multiplayer/tests/Multiplayer_SimpleNetworkLevelEntity.py (hunk 1 above)
Comment (AMZN-puvvadar) [reply AutomatedTesting/Gem/PythonTests/Multiplayer/tests/Multiplayer_SimpleNetworkLevelEntity.py:61]: Are there any concerns around flakiness with the timer?

File: AutomatedTesting/Gem/PythonTests/Multiplayer/tests/Multiplayer_SimpleNetworkLevelEntity.py (hunk 2 above)
Comment (AMZN-Gene) [reply AutomatedTesting/Gem/PythonTests/Multiplayer/tests/Multiplayer_SimpleNetworkLevelEntity.py]: fixed!

File: AutomatedTesting/Gem/PythonTests/Multiplayer/tests/Multiplayer_SimpleNetworkLevelEntity.py (hunk 1 above)
Comment (AMZN-Gene) [reply AutomatedTesting/Gem/PythonTests/Multiplayer/tests/Multiplayer_SimpleNetworkLevelEntity.py]: Good question! This particular error was happening before the handshake occurred, so 1.0 second is too long if anything. Dropped this down to 0 seconds... 1 second saved :D

File: Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.h [hunk 3]
```diff
@@ -184,7 +184,7 @@ namespace Multiplayer
         double m_serverSendAccumulator = 0.0;
         float m_renderBlendFactor = 0.0f;
         float m_tickFactor = 0.0f;
-        bool m_spawnNetboundEntities = true;
+        bool m_spawnNetboundEntities = false;
```
Comment (kberg-amzn) [inline Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.h:187]: If you ctrl-G, escape, and ctrl-G again does this properly get reset to false?

**Review by AMZN-puvvadar — APPROVED** [review]
Seems fine if the timer usages are considered stable.

File: Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.h (hunk 3 above)
Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.h:187]: The editor is a client so it'll stay false. Escaping gamemode will turn the agent type from "client" back to "unintialized" though.

**Review by lmbr-pip — APPROVED** [review]
LG.

**Pull request merged**